addons/account/models/account_account.py

```diff
@@ -1017,8 +1020,14 @@ def _adapt_parent_account_group(self):
         """
         if self.env.context.get('delay_account_group_sync'):
             return
-        if not self:
+        if company:
+            company_ids = company.ids
+        else:
+            company_ids = self.company_id.ids
+
+        if not company_ids:
             return
+        cond = SQL('child.company_id IN %s', tuple(company_ids))
```

Comment:
Why is this outside of the main query? :thinking:

Reply:
Was conditional at some point